### PR TITLE
feat: Add support for "transit_gateway" and "customer_gateway" AWS resources

### PR DESCRIPTION
**Pull request by dbarranco**
Hello!
This PR solves https://github.com/GoogleCloudPlatform/terraformer/issues/279.

As you can see in the "transit_gateway" file, there are some transit gateway attachments that were needed to comment out, as the Terraform plugin doesn't support them.

Thanks!

### CODE REVIEW

File: providers/aws/customer_gateway.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2020 VMware, Inc.
```
Comment (sergeylanzman) [inline providers/aws/customer_gateway.go]: please change text as in other files

File: providers/aws/transit_gateway.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2020 VMware, Inc.
```
Comment (sergeylanzman) [inline providers/aws/transit_gateway.go]: same

File: providers/aws/transit_gateway.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	}
+	return p.Err()
+}
+
+func (g *TransitGatewayGenerator) getTransitGatewayAttachments(svc *ec2.Client) error {
+	p := ec2.NewDescribeTransitGatewayAttachmentsPaginator(svc.DescribeTransitGatewayAttachmentsRequest(&ec2.DescribeTransitGatewayAttachmentsInput{}))
+	for p.Next(context.Background()) {
+		for _, tgwa := range p.CurrentPage().TransitGatewayAttachments {
+			switch tgwa.ResourceType {
+			// terraform@v0.12.18 doesn't support this resource type
```
Comment (sergeylanzman) [inline providers/aws/transit_gateway.go]: ResourceType it's AWS SDK type, why terraform relevant for this?

**Comment by sergeylanzman** [comment]
Thank you!
Can you please delete vendor file from PE and squash to one commit?

File: providers/aws/transit_gateway.go (hunk 3 above)
Comment (dbarranco) [reply providers/aws/transit_gateway.go]: This was the error trace I got once I added the `aws_ec2_transit_gateway_vpn_attachment` resource:

```
2020/01/13 13:25:29 aws importing... transit_gateway
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_route_table.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpc_attachment.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpc_attachment.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpc_attachment.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpc_attachment.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpc_attachment.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpc_attachment.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpc_attachment.tfer--xxx
2020/01/13 13:25:36 Refreshing state... aws_ec2_transit_gateway_vpn_attachment.tfer--xxx
panic: unknown resource type aws_ec2_transit_gateway_vpn_attachment

goroutine 536 [running]:
github.com/hashicorp/terraform/plugin.(*GRPCProvider).getResourceSchema(0xc0000c60e0, 0x3450d70, 0x26, 0x0, 0x0)
        /Users/dbarranco/go/pkg/mod/github.com/hashicorp/terraform@v0.12.18/plugin/grpc_provider.go:92 +0x129
github.com/hashicorp/terraform/plugin.(*GRPCProvider).ReadResource(0xc0000c60e0, 0x3450d70, 0x26, 0x3939400, 0xc0006f8180, 0x2e087e0, 0xc001322d20, 0x5449c30, 0x0, 0x0, ...)
        /Users/dbarranco/go/pkg/mod/github.com/hashicorp/terraform@v0.12.18/plugin/grpc_provider.go:321 +0xa3
github.com/GoogleCloudPlatform/terraformer/terraform_utils/provider_wrapper.(*ProviderWrapper).Refresh(0xc0004463c0, 0xc000447310, 0xc0004472c0, 0x2, 0xc00058a2a0, 0x68)
```

Doesn't it seems to be related with the version of Terraform that the project is currently using?

**Comment by dbarranco** [comment]
Thanks for the quick review.
The copyright statements were imposed to me by VMware, so I'm not sure what to do about that :(

What did you do in the past with this kind of changes in files? If you have some doc. pointers that'd help me to have some internal discussions with the VMware folks.

**Comment by dbarranco** [comment]
> Can you please delete vendor file from PE and squash to one commit?

Will do that, thanks

File: providers/aws/transit_gateway.go (hunk 3 above)
Comment (sergeylanzman) [reply providers/aws/transit_gateway.go]: I think it's related your aws provider version.

Comment (meshuga) [reply providers/aws/transit_gateway.go]: `aws_ec2_transit_gateway_vpn_attachment ` is a data source, not a resource.

Comment (dbarranco) [reply providers/aws/transit_gateway.go]: crap, it is true.
The only current Terraform resource supported at the moment are the vpc attachments.
Will change that.

**Comment by dbarranco** [comment]
btw @meshuga there's another interesting topic around the Transit gateway route tables:

```
resource "aws_ec2_transit_gateway_route_table" "tfer--example-id" {
  transit_gateway_id = "${data.terraform_remote_state.transit_gateway.outputs.aws_ec2_transit_gateway_tfer--example-id}"
}

```

I guess it is not worth at all to dump this resource? As if I re-create the Transit gateway with terraform the route table will be automatically created.

**Comment by meshuga** [comment]
You mean default route tables will be automatically created? Apart from default tables, a user can create its own tables. In that case, you should filter out default tables and generate only resources for custom ones.

Regarding the copyright statement, I think there's nothing much that can be done since that's your company policy. On a side note, I still remember an early story of Vert.x framework...

**Comment by dbarranco** [comment]
Nice, now the default route tables that are automatically created/populated when new attachments are connected to the transit_gateway are not included in the Terraform manifests created by Terraformer

**Comment by sergeylanzman** [comment]
@dbarranco Can you resolve licensing comment issue?

**Comment by meshuga** [comment]
@dbarranco @sergeylanzman It looks cooperation in OSS development between VMware and Google is quite complicated. Is there anything that can be done to ease that?

**Comment by dbarranco** [comment]
hello, @meshuga @sergeylanzman 

Sorry I didn't reply before to this PR, I got other urgent priorities.
With regards to the license conflicts.... I'm not really sure what can I do here. I'll try to reach some folks in VMware to see if they faced something similar in the past.

Not sure if you could help me from your side :( 
thanks!

**Comment by dbarranco** [comment]
ok, contacted a few VMware folks and I didn't know we have a policy to respect the project copyright notices.

so... am I done with this? :)
Thanks

**Pull request merged**